src/tchannel_bridge.js

```diff
@@ -0,0 +1,128 @@
+     * a the handler's context with a span.
+     **/
+    tracedHandler(handlerFunc: any, options: startSpanArgs = {}): Function {
+        return (context, request, headers, body, callback) => {
+            let operationName = options.operationName || request.arg1;
+            let span = this._getSpanFromTChannelRequest(operationName, headers);
+            // In theory may overwrite tchannel span, but thats what we want anyway.
+            context.span = span;
+
+            handlerFunc(context, request, headers, body, callback);
+
+            // when handlerFunc returns then finish the span.
+            span.finish();
+        };
+    }
+
+    /**
+     * A function that wraps a json, or thrift encoded channel, in order to populate
+     * the outgoing headers with trace context, and baggage information.
+     *
+     * @param {Object} channel - the encoded channel to be wrapped for tracing.
+     * @param {Object} context - A context that contains the outgoing span to be seralized.  If the context does not contain a span then a new root span is created.
+     * @returns {Object} channel - the trace wrapped channel.
+     * */
+    tracedChannel(channel: any, context: any = {}): any {
+        let wrappedRequestFunc = channel.request.bind(channel);
+        channel.request = (requestOptions) => {
+            // We set the parent to a span with trace_id zero, so that tchanne's
+            // outgoing tracing frame also has a trace id of zero.
+            // This forces other tchannel implementations to rely on the headers for the trace context.
+            requestOptions.parent = { span: this._getTChannelParentSpan() };
+            let channelRequest = wrappedRequestFunc(requestOptions);
+            let wrappedSend = channelRequest.send.bind(channelRequest);
+            channelRequest.send = (endpoint, headers, body, wrappedCallback) => {
+                headers = headers || {};
+
+                // populate headers with span state
+                if (!context.span) {
+                    context.span = this._tracer.startSpan(endpoint);
+                }
+                this._saveSpanStateToCarrier(context.span, headers);
+
+                // wrap callback so that span can be finished as soon as the response is received
+                let callback = () => {
+                    context.span.finish();
+                    wrappedCallback.apply(this, arguments);
+                }
+
+                return wrappedSend(endpoint, headers, body, callback);
+            };
+
+            return channelRequest;
+        };
+
+        return channel;
+    }
+
+    _getTChannelParentSpan(): TChannelSpan {
+        return {
+            id: [0, 0],
+            traceid: [0, 0],
+            parentid: [0, 0],
+            flags: 0
+        };
+    }
+
+    _getSpanFromTChannelRequest(operationName: string, headers: any = {}, options: any = {}): Span {
```

Comment:
* nit: rename `_extractSpan`
* why do we default `headers` in the signature?  if null is passed, would that defaulting still work?
* the bridge is the only code that actually calls this method, so why do we need the `options`?